### ISSUE TITLE
Coco D1 Model is not detecting anything with Inference.py

### ISSUE THREAD

**Issue by SiBensberg** [xuannianz/EfficientDet#105]
Hi,
I trained my Model with the following command
`python3 train.py --snapshot imagenet --phi 1 --weighted-bifpn --gpu 0,1 --random-transform --epochs 150 --compute-val-loss --freeze-backbone --batch-size 20 --steps 1000 coco /coco2017`

I trained for 130 Epochs and Tensorboard reports me following Values:
![grafik](https://user-images.githubusercontent.com/55738346/77083677-2d3fff80-69fe-11ea-8038-8d4cb96568fc.png)
![grafik](https://user-images.githubusercontent.com/55738346/77083765-4cd72800-69fe-11ea-9aad-5e3b05f7afbb.png)

looks good to me. 
Now i wanted to test the "half"-trained model, but Inference.py not detecting anything. Even with Threshold 0.05. Why?

```

    phi = 1
    weighted_bifpn = True
    model_path = 'checkpoints/2020-03-10/coco_130_1.6978_1.6937.h5'
    image_sizes = (512, 640, 768, 896, 1024, 1280, 1408)
    image_size = image_sizes[phi]
    classes = [
        'person',
        'bicycle',
        'car',
        'motorcycle',
        'airplane',
        'bus',
        'train',
        'truck',
        'boat',
        'traffic light',
        'fire hydrant',
        'stop sign',
        'parking meter',
        'bench',
        'bird',
        'cat',
        'dog',
        'horse',
        'sheep',
        'cow',
        'elephant',
        'bear',
        'zebra',
        'giraffe',
        'backpack',
        'umbrella',
        'handbag',
        'tie',
        'suitcase',
        'frisbee',
        'skis',
        'snowboard',
        'sports ball',
        'kite',
        'baseball bat',
        'baseball glove',
        'skateboard',
        'surfboard',
        'tennis racket',
        'bottle',
        'wine glass',
        'cup',
        'fork',
        'knife',
        'spoon',
        'bowl',
        'banana',
        'apple',
        'sandwich',
        'orange',
        'broccoli',
        'carrot',
        'hot dog',
        'pizza',
        'donut',
        'cake',
        'chair',
        'couch',
        'potted plant',
        'bed',
        'dining table',
        'toilet',
        'tv',
        'laptop',
        'mouse',
        'remote',
        'keyboard',
        'cell phone',
        'microwave',
        'oven',
        'toaster',
        'sink',
        'refrigerator'
        'book',
        'clock',
        'vase',
        'scissors',
        'teddy bear',
        'hair drier',
        'toothbrush'
    ]
    num_classes = len(classes)
    score_threshold = 0.05
```

Just the Picture without any boxes is shown and no error gets displayed.

Thanks for your great work.

Greetings
SiBensberg

**Comment by SiBensberg** [comment]
so I tried different phis and weighted-bifpn=false and nothing seems to work.

**Comment by xuannianz** [comment]
@SiBensberg , sorry for the late reply. Have you found the answers?

**Comment by SiBensberg** [comment]
Hi @xuannianz 
sadly not, in the meantime I used an older Version of your code as a workaround. It works like charm, but sadly not with multi GPU. 
I guess something went wrong with the model, because messing with the classes list in "inference.py" gets me a failure with the older version. With the newer version it just outputs an empty picture. 
I saw you updated the code today with tf2 support. Is it still possible to export a frozen graph with models trained with tf2?

**Comment by hareld10** [comment]
Hi,

It's really weird that the demo code- just don't predict anything. (on any pretrained model) 

I saw you marked it as it won't be fix, but it is important enough I think..

**Comment by SiBensberg** [comment]
I tried #108 But sadly it still does not detect anything. Do I have to retrain my model with the new environment? 

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_tflow_select             2.3.0                       mkl  
absl-py                   0.9.0                    py36_0  
asn1crypto                1.3.0                    py36_0  
astor                     0.8.0                    py36_0  
attrs                     19.3.0                     py_0  
blas                      1.0                         mkl  
blinker                   1.4                      py36_0  
c-ares                    1.15.0            h7b6447c_1001  
ca-certificates           2020.1.1                      0  
cachetools                3.1.1                      py_0  
certifi                   2019.11.28               py36_1  
cffi                      1.14.0           py36h2e261b9_0  
chardet                   3.0.4                 py36_1003  
click                     7.1.1                      py_0  
cryptography              2.8              py36h1ba5d50_0  
freetype                  2.9.1                h8a8886c_1  
gast                      0.2.2                    py36_0  
google-auth               1.11.2                     py_0  
google-auth-oauthlib      0.4.1                      py_2  
google-pasta              0.2.0                      py_0  
grpcio                    1.27.2           py36hf8bcb03_0  
h5py                      2.10.0           py36h7918eee_0  
hdf5                      1.10.4               hb1b8bf9_0  
idna                      2.9                        py_1  
importlib_metadata        1.5.0                    py36_0  
intel-openmp              2020.0                      166  
jpeg                      9b                   h024ee3a_2  
keras                     2.2.5                    py36_1    conda-forge
keras-applications        1.0.8                      py_0  
keras-preprocessing       1.1.0                      py_1  
ld_impl_linux-64          2.33.1               h53a641e_7  
libedit                   3.1.20181209         hc058e9b_0  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
libgpuarray               0.7.6             h14c3975_1003    conda-forge
libpng                    1.6.37               hbc83047_0  
libprotobuf               3.11.4               hd408876_0  
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.1.0                h2733197_0  
mako                      1.1.0                      py_0    conda-forge
markdown                  3.1.1                    py36_0  
markupsafe                1.1.1            py36h8c4c3a4_1    conda-forge
mkl                       2020.0                      166  
mkl-service               2.3.0            py36he904b0f_0  
mkl_fft                   1.0.15           py36ha843d7b_0  
mkl_random                1.1.0            py36hd6b4f25_0  
more-itertools            8.2.0                      py_0  
ncurses                   6.2                  he6710b0_0  
numpy                     1.18.1           py36h4f9e942_0  
numpy-base                1.18.1           py36hde5b4d6_1  
oauthlib                  3.1.0                      py_0  
olefile                   0.46                     py36_0  
openssl                   1.1.1e               h7b6447c_0  
opt_einsum                3.1.0                      py_0  
packaging                 20.3                       py_0  
pillow                    6.2.0            py36h34e0f95_0  
pip                       20.0.2                   py36_1  
pluggy                    0.13.1                   py36_0  
progressbar2              3.37.1                   py36_0  
protobuf                  3.11.4           py36he6710b0_0  
py                        1.8.1                      py_0  
pyasn1                    0.4.8                      py_0  
pyasn1-modules            0.2.7                      py_0  
pycparser                 2.20                       py_0  
pygpu                     0.7.6           py36hc1659b7_1000    conda-forge
pyjwt                     1.7.1                    py36_0  
pyopenssl                 19.1.0                   py36_0  
pyparsing                 2.4.6                      py_0  
pysocks                   1.7.1                    py36_0  
pytest                    5.4.1                    py36_0  
pytest-runner             5.2                        py_0  
python                    3.6.10               hcf32534_1  
python-utils              2.3.0                    py36_0  
python_abi                3.6                     1_cp36m    conda-forge
pyyaml                    5.3.1            py36h8c4c3a4_0    conda-forge
readline                  8.0                  h7b6447c_0  
requests                  2.23.0                   py36_0  
requests-oauthlib         1.3.0                      py_0  
rsa                       4.0                        py_0  
scipy                     1.4.1            py36h0b6359f_0  
setuptools                46.1.1                   py36_0  
six                       1.14.0                   py36_0  
sqlite                    3.31.1               h7b6447c_0  
tensorboard               2.1.0                     py3_0  
tensorflow                2.1.0           mkl_py36h23468d9_0  
tensorflow-base           2.1.0           mkl_py36h6d63fb7_0  
tensorflow-estimator      2.1.0              pyhd54b08b_0  
termcolor                 1.1.0                    py36_1  
theano                    1.0.3            py36hfc679d8_1    conda-forge
tk                        8.6.8                hbc83047_0  
toolchain                 2.4.0                         0  
toolchain_c_linux-64      2.4.0                         0  
toolchain_cxx_linux-64    2.4.0                         0  
tqdm                      4.28.1           py36h28b3542_0  
urllib3                   1.25.8                   py36_0  
wcwidth                   0.1.8                      py_0  
werkzeug                  1.0.0                      py_0  
wheel                     0.34.2                   py36_0  
wrapt                     1.12.1           py36h7b6447c_1  
xz                        5.2.4                h14c3975_4  
yaml                      0.2.2                h516909a_1    conda-forge
zipp                      2.2.0                      py_0  
zlib                      1.2.11               h7b6447c_3  

```

**Comment by hareld10** [comment]
@SiBensberg Just open fresh docker - install requirements and enjoy.
You surely don't need to retrain - that's not the issue

**Comment by SiBensberg** [comment]
I made a fresh conda environment. What puzzles me is, it works good with the D1 weights from @xuannianz. Sadly it does not on my own trained model.

**Comment by SiBensberg** [comment]
I just threw away this model. I have a similiar Problem with a newer model tho. #129

**Comment by cyrilzakka** [comment]
Have the same issue as well!
